dev/tools/controllerbuilder/pkg/codegen/typegenerator.go

```diff
@@ -362,6 +363,22 @@ func GoNameForProtoMessage(msg protoreflect.MessageDescriptor) string {
 
 	fullName = strings.TrimPrefix(fullName, string(msg.ParentFile().FullName()))
 	fullName = strings.TrimPrefix(fullName, ".")
+	// Ensure acronyms in type names are also handled.
+	parts := strings.Split(fullName, ".")
+	for i, part := range parts {
+		partInSnakeCase := text.AsSnakeCase(part)
+		tokens := strings.Split(partInSnakeCase, "_")
+		for j, token := range tokens {
+			if IsAcronym(token) {
+				token = strings.ToUpper(token)
+			} else {
+				token = strings.Title(token)
+			}
+			tokens[j] = token
+		}
+		parts[i] = strings.Join(tokens, "")
+	}
+	fullName = strings.Join(parts, ".")
```

Comment:
nit: it seems the last 3 lines can be combined into `return strings.Join(parts, "_")`.

Reply:
Done.